test/app-luatest/init_script_test.lua

```diff
@@ -0,0 +1,144 @@
+-- The test tests init script.
+
+local t = require('luatest')
+local treegen = require('luatest.treegen')
+local justrun = require('luatest.justrun')
+
+local g = t.group()
+
+-- Access to box.cfg from init script.
+g.test_access_box_cfg_init_script= function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_access_box_cfg_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+box.cfg({memtx_memory = 107374182})
+print('box.cfg.memtx_memory ' .. tostring(box.cfg.memtx_memory))
+os.exit()
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.stdout, 'box.cfg.memtx_memory 107374182')
+end
+
+-- The testcases tests insert tuples using fiber in init script.
+g.test_insert_tuples_fiber_init_script = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_insert_tuples_fiber_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+local fiber = require('fiber')
+local json = require('json')
+
+box.cfg()
+local space = box.schema.space.create('tweedledum')
+space:create_index('primary', { type = 'hash' })
+
+function do_insert()
+    space:insert({1, 2, 4, 8})
+end
+
+fiber.create(do_insert)
+
+print(json.encode(space:select(nil, { limit = 4 })))
+
+os.exit()
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.stdout, '[[1,2,4,8]]')
```

Comment:
Added:

```diff
diff --git a/test/app-luatest/init_script_test.lua b/test/app-luatest/init_script_test.lua
index e591e24d72..73be74ea12 100644
--- a/test/app-luatest/init_script_test.lua
+++ b/test/app-luatest/init_script_test.lua
@@ -17,8 +17,8 @@ os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
-    t.assert_equals(res.stdout, 'box.cfg.memtx_memory 107374182')
+    t.assert_equals(res.exit_code, 0, 'exit code')
+    t.assert_equals(res.stdout, 'box.cfg.memtx_memory 107374182', 'stdout')
 end
 
 -- The testcases tests insert tuples using fiber in init script.
@@ -47,8 +47,8 @@ os.exit(0)
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.stdout, '---\
 - [1, 2, 4, 8]\
-...')
-    t.assert_equals(res.exit_code, 0)
+...', 'stdout')
+    t.assert_equals(res.exit_code, 0, 'exit code')
 end
 
 -- The testcase tests insert from detached fiber.
@@ -73,11 +73,11 @@ os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.exit_code, 0, 'exit code')
     t.assert_equals(res.stdout, [[
 ---
 - [1, 2, 4, 8]
-...]])
+...]], 'stdout')
 end
 
 -- The testcase tests insert from an init script.
@@ -99,10 +99,9 @@ os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.exit_code, 0, 'exit code')
     t.assert_equals(res.stdout, '\
-[2, 4, 8, 16]'
-)
+[2, 4, 8, 16]', 'stdout')
 end
 
 -- The testcase checks that `math.floor` is reachable in the init
@@ -120,11 +119,11 @@ os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.exit_code, 0, 'exit code')
     t.assert_equals(res.stdout, [[
 0
 0
-1]])
+1]], 'stdout')
 end
 
 -- A test case for tarantool/tarantool#53.
@@ -142,5 +141,5 @@ os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.exit_code, 0, 'exit code')
 end
```

These messages are not visible nor in TAP:

```
Tarantool version is 3.3.0-entrypoint-188-gf3ecdb1b40
TAP version 13
1..6
# Started on Fri Nov  1 21:59:49 2024
# Starting group: app-luatest.init_script
ok     1        app-luatest.init_script.test_access_box_cfg_init_script
ok     2        app-luatest.init_script.test_insert_tuples_fiber_init_script
ok     3        app-luatest.init_script.test_insert_tuples_and_select_init_script
ok     4        app-luatest.init_script.test_insert_tuples_init_script
ok     5        app-luatest.init_script.test_require_math_floor_init_script
ok     6        app-luatest.init_script.test_gh_53
# Ran 6 tests in 1.410 seconds, 6 succeeded, 0 failed
```

nor in `-v`:

```
Tarantool version is 3.3.0-entrypoint-188-gf3ecdb1b40
Started on Fri Nov  1 22:00:20 2024
    app-luatest.init_script.test_access_box_cfg_init_script ... (0.210s) Ok
    app-luatest.init_script.test_insert_tuples_fiber_init_script ... (0.211s) Ok
    app-luatest.init_script.test_insert_tuples_and_select_init_script ... (0.216s) Ok
    app-luatest.init_script.test_insert_tuples_init_script ... (0.226s) Ok
    app-luatest.init_script.test_require_math_floor_init_script ... (0.154s) Ok
    app-luatest.init_script.test_gh_53 ... (0.173s) Ok
=========================================================
Ran 6 tests in 1.195 seconds, 6 succeeded, 0 failed
```

Why do you need these messages in assertions?

Reply:
I also found that when we have a good diagnostic about compared values in case of a failure, these messages just pollute the testing code. I don't write them most of time (unless situations, when I know I'm able to make things more clear -- usually in testing helpers and not a test case itself).

Reply:
My bad, then. Just the habit with TAP test format.